client/Elements/Components/CreateTopic/StepMessageRetention.patternfly.tsx

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+import React, { useState } from 'react';
+import '@patternfly/react-core/dist/styles/base.css';
+import {
+  Flex,
+  FlexItem,
+  Form,
+  FormGroup,
+  Radio,
+  Select,
+  SelectOption,
+  SelectVariant,
+  TextContent,
+  Text,
+  TextVariants,
+  Touchspin,
+} from '@patternfly/react-core';
+import './CreateTopicWizard.patternfly.css';
+
+interface IStepMessageRetention {
+  setMsgRetentionValue: (value: number) => void;
+}
+
+export const StepMessageRetention: React.FC<IStepMessageRetention> = ({
+  setMsgRetentionValue,
+}) => {
+  const [radioBtnDay, setRadioBtnDay] = useState(false);
+  const [radioBtnWeek, setRadioBtnWeek] = useState(false);
+  const [radioBtnMonth, setRadioBtnMonth] = useState(false);
+  const [radioBtnCustomTime, setRadioBtnCustomTime] = useState(false);
```

Comment:
Would be better to defined valid Retention options (DAY, WEEK, MONTH, CUSTOM) and then use a single state:

`const [currentPeriod, setCurrentPeriod] = React.useState<string>(RetentionOption.MONTH);`